client.go

```diff
@@ -168,7 +147,7 @@ func shallowDefaultTransport() *http.Transport {
 	}
 }
 
-func getUserAuth(ctx context.Context, config *Config, endpoint *Endpoint) (*Config, error) {
```

Comment:
not directly related- but I went ahead and made this pass by value. Passing by pointer allows the caller to modify the data. Passing/returning by value allow the caller to check the error result and accept (or decline) the config object based on the error.